docs/testing_your_changes.md

```diff
@@ -0,0 +1,52 @@
+## Location of tests
+
+OpenFGA strives to maintain good code coverage with tests so that the possibility of regressions is reduced.
+
+The purpose of this guide is to inform contributors where to write new tests.
+
+## 1. Storage layer
+
+1. Any change to the OpenFGA storage interface must have an integration test in pkg/storage/test/storage.go. This is so that anyone writing their own storage adapter can have confidence that it is correct.
+
+## 2. API layer
+
+1. All APIs use an underlying `command` object. These commands must have
+   1. unit tests. For example, Write API is backed by Write Command, which has unit tests in pkg/server/commands/write_test.go. These should target mocked dependencies to cover the majority of the business logic. If you are changing business logic, add a test here.
+   2. **Exported** integration tests in pkg/server/test/server.go. These take in an unmocked `datastore` dependency. The goal of these is to test that the command and the underlying datastore work well. This is so that anyone who writes their own implementation of the datastore can run these tests by doing
```

Comment:
```suggestion
   2. **Exported** integration tests in `pkg/server/test/server.go`. These take in an unmocked `datastore` dependency. The goal of these is to test that the command and the underlying datastore work well. This is so that anyone who writes their own implementation of the datastore can run these tests by doing
```